tests/phpunit/tests/html-api/wpHtmlTagProcessor.php

```diff
@@ -2875,4 +2875,35 @@ public function insert_after( $new_html ) {
 			'Should have properly applied the update from in front of the cursor.'
 		);
 	}
+
+	/**
+	 * @ticket TBD
+	 *
+	 * @covers ::remove_class
+	 */
+	public function test_remove_class_removes_leading_white_space() {
+		$processor = new WP_HTML_Tag_Processor( '<div class="remove keep">' );
+		$processor->next_tag();
+		$processor->remove_class( 'remove' );
+		$this->assertSame(
+			'<div class="keep">',
+			$processor->get_updated_html(),
+			'::remove_class did not correctly clean whitespace from removed class.'
+		);
+	}
```

Comment:
I strongly recommend we not encode this behavior in a test as it represents an incidental detail that isn't relevant to the processing. It's true that other tests will break when the behavior changes, but that's because the other tests are written without a proper HTML parser to compare semantic equality.

This test actually enforces a behavior we specifically want to avoid enforcing, since whitespace doesn't matter in the `class` attribute. We should have a test that says, `test_ensures_that_removing_or_leaving_whitespace_are_both_fine()` but we can't test that in any way I can think of